sdk/core/src/test/java/com/klaviyo/core/networking/KlaviyoNetworkMonitorTest.kt

```diff
@@ -130,6 +134,7 @@ internal class KlaviyoNetworkMonitorTest : BaseTest() {
         netCallbackSlot.captured.onLost(mockk())
 
         assertEquals(6, callCount)
+        expectedNetworkConnection = true
```

Comment:
The reason for this is pretty interesting - the assertion for this test will still run even **after** the test case finishes, which I believe exposes a pitfall in our `set listener` design here. Without this reset to `true`, my new unit test would fail `assert(it == expectedNetworkConnection)` since we start the test by adding a connection. Changing the expectation to `true` essentially resets this to the same state as the start.

Reply:
If I follow that correctly, this basically serves the purpose of resetting the test class's expectations prior to the next test? Would it work to put this in a setup or teardown function instead?

Reply:
I ended up changing this to instead remove the observer at the end of the test - it is a bit tricky since we declare the count in the test locally but modify the singleton network monitor. I think it makes sense that at the end of the individual test we remove the observer.